pkg/solana/client/client.go

```diff
@@ -157,7 +161,9 @@ func (c *Client) GetTransaction(ctx context.Context, txHash solana.Signature, op
 		opts.Commitment = c.commitment
 	}
 
-	v, err, _ := c.requestGroup.Do("GetTransaction", func() (interface{}, error) {
+	// Use txHash in the key so different signatures won't be merged on concurrent calls.
+	key := fmt.Sprintf("GetTransaction(%s)", txHash.String())
```

Comment:
It's possible to pass custom opts. We should include them into the key or remove it from the function arguments.
I'm leaning towards second option as the method is only used in one test

Reply:
Nice! Adding this

Reply:
Had to make some changes in the new `multi_client` to remove the extra param